Sources/SimpleCurrencyConverter/ExchangeRatesNetworkManager.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+
+public enum ExchangeRatesError: Error {
+    case failedToGetResponse
+    case failedToConfigureURL
+    case invalidResponse
+    case failedToDecodeResponse
+}
+ 
+enum Keys {
+    static let base = "base"
+    static let symbols = "symbols"
+    static let rates = "rates"
+    static let from = "from"
+    static let to = "to"
+    static let amount = "amount"
+    static let result = "result"
+    
+    static let httpMethod = "GET"
+    static let httpHeader = "apikey"
+}
+
+final class ExchangeRatesNetworkManager {
+    
+    private let session: URLSession
+    
+    init(session: URLSession) {
+        self.session = session
+    }
+    
+    func requestExchangeRate(apiKey: String?, url: URL, base: Currency, target: [Currency]) async throws -> Result<[Currency: Double], Error> {
```

Comment:
@ts-dzmitry-kurski Is `apiKey` mandatory?
If yes, can you pass `apiKey` as non-optional?